File: practice/data_structures/linked_lists/get_node_value/get_node_value.cpp
/*
  Get Nth element from the end in a linked list of integers
  Number of elements in the list will always be greater than N.
  Node is defined as 
  struct Node
  {
     int data;
     struct Node *next;
  }
*/

int GetNode(Node * i_head, int i_positionFromTail)
{
    Node *head1 = i_head, *head2 = i_head;
    
    int cnt = 0;
    while (head1 && (cnt < i_positionFromTail))
    {
        cnt++;
        head1 = head1->next;
    }
    
    if (cnt != i_positionFromTail)
    {
        //error
        return -1;
    }
    
    while (head1->next)
    {
        head1 = head1->next;
        head2 = head2->next;
    }
    
    return head2->data;
}

File: practice/data_structures/linked_lists/get_node_value/get_node_value.py
"""
 Get Node data of the Nth Node from the end.
 head could be None as well for empty list
 Node is defined as
 
 class Node(object):
 
   def __init__(self, data=None, next_node=None):
       self.data = data
       self.next = next_node

 return back the node data of the linked list in the below method.
"""

def GetNode(i_head, i_position):
    head1 = i_head
    head2 = i_head
    
    cnt = 0
    while head1 and cnt < i_position:
        head1 = head1.next;
        cnt += 1
        
    while head1.next:
        head1 = head1.next
        head2 = head2.next
        
    return head2.data

